src/components/ComputeModal.js

```diff
@@ -393,8 +410,8 @@ export const ComputeModalBase = ({ onDismiss, onSuccess, runtimes, persistentDis
 
   const getDesiredEnvironmentConfig = () => {
     const { persistentDisk: existingPersistentDisk, runtime: existingRuntime } = getExistingEnvironmentConfig()
-    const cloudService = sparkMode ? cloudServices.DATAPROC : cloudServices.GCE
-    const desiredNumberOfWorkers = sparkMode === 'cluster' ? computeConfig.numberOfWorkers : 0
+    const cloudService = isDataproc(runtimeType) ? cloudServices.DATAPROC : cloudServices.GCE
```

Comment:
You could have gone a different way with this refactoring and have had the different instances of runtimeType implement methods to provide this information so that you don't need conditionals. In this case, they would each have a "getCloudService" method and a "isDataproc" method, and you would not need "isGce", "isDataproc", and "isDataprocCluster". This is how we went with `tools` in `notebook-utils`, and it is more extensible as additional properties of types get added.

I'm not really advocating that you rework this PR, but just mentioning it as an alternative approach, especially if the properties of a runtimeType continue to grow.

Reply:
Thank you for the thoughts. I started with a scheme similar to what you described but I didn't like the verbosity it resulted in at call sites. Also, having `isDataproc` as a property did not seem to be scalable as we add more cloud services like `Azure VM` where we'd also need `isAzureVm`, etc.